src/test/conftest.py

```diff
@@ -1,9 +1,39 @@
+import dataclasses
+from typing import Type, TypeVar
+
 import pytest
 
 from storage.db_setup import DbSetup
 from test.common_helper import clear_test_tables, setup_test_tables
 
 
+def merge_markers(request, name: str, dtype: Type) -> TypeVar('T'):
```

Comment:
Good catch. No idea if this would have worked.

But your solution is better anyways since it shows the relation between `dytpe` and the return value.